src/libexpr/primops.cc

```diff
@@ -2221,6 +2235,9 @@ void EvalState::createBaseEnv()
     addPrimOp("__sub", 2, prim_sub);
     addPrimOp("__mul", 2, prim_mul);
     addPrimOp("__div", 2, prim_div);
+    addPrimOp("__bin_and", 2, prim_bin_and);
+    addPrimOp("__bin_or", 2, prim_bin_or);
+    addPrimOp("__bin_xor", 2, prim_bin_xor);
```

Comment:
The only remaining issue is that the naming is inconsistent with all other builtin functions, which use camelCase. Maybe `bitAnd` etc.?